File: 1491-average-salary-excluding-the-minimum-and-maximum-salary/1491-average-salary-excluding-the-minimum-and-maximum-salary.py
class Solution:
    def average(self, salary: List[int]) -> float:
        min_sal, max_sal = float('inf'), float('-inf')
        sum_ = 0
        for sal in salary:
            min_sal = min(min_sal, sal)
            max_sal = max(max_sal, sal)
            sum_ += sal
        return (sum_ - min_sal - max_sal) / (len(salary) - 2)
        
        
#     def average(self, salary: List[int]) -> float:
#         """
        
#         """
#         min_sal = min(salary)
#         max_sal = max(salary)        
#         return (sum(salary) - min_sal - max_sal) / (len(salary) - 2)

File: 1491-average-salary-excluding-the-minimum-and-maximum-salary/1491-average-salary-excluding-the-minimum-and-maximum-salary.java
class Solution {
    public double average(int[] salary) {
        int minSal = Integer.MAX_VALUE;
        int maxSal = Integer.MIN_VALUE;
        int sum = 0;
        
        for (int sal: salary) {
            minSal = Math.min(minSal, sal);
            maxSal = Math.max(maxSal, sal);
            sum += sal;
        }
        
        return (double) (sum - minSal - maxSal) / (salary.length - 2);
    }
}